package/MDAnalysis/coordinates/timestep.pyx

```diff
@@ -0,0 +1,982 @@
+        except NoDataError:
+            pass
+
+
+        try:
+            other._reader = weakref.ref(ts._reader())
+        except TypeError:  # TypeError from calling None weakref
+            pass
+
+        ts.data = copy.deepcopy(other.data)
+
+        return ts
+
+    @classmethod
+    def from_coordinates(cls,
+                         positions=None,
+                         velocities=None,
+                         forces=None,
+                         **kwargs):
+        """Create an instance of this Timestep, from coordinate data
+
+        Can pass position, velocity and force data to form a Timestep.
+
+        .. versionadded:: 0.11.0
+        """
+        has_positions = positions is not None
+        has_velocities = velocities is not None
+        has_forces = forces is not None
+
+        lens = [len(a) for a in [positions, velocities, forces]
+                if a is not None]
+        if not lens:
+            raise ValueError("Must specify at least one set of data")
+        n_atoms = max(lens)
+        # Check arrays are matched length?
+        if not all(val == n_atoms for val in lens):
+            raise ValueError("Lengths of input data mismatched")
+
+        ts = cls(n_atoms,
+                 positions=has_positions,
+                 velocities=has_velocities,
+                 forces=has_forces,
+                 **kwargs)
+        if has_positions:
+            ts.positions = positions
+        if has_velocities:
+            ts.velocities = velocities
+        if has_forces:
+            ts.forces = forces
+
+        return ts
+
+    def __eq__(self, other):
+        """Compare with another Timestep
+
+        .. versionadded:: 0.11.0
+        """
+        if not isinstance(other, Timestep):
+            return False
+
+        if not self.frame == other.frame:
+            return False
+
+        if not self.n_atoms == other.n_atoms:
+            return False
+
+        if not self.has_positions == other.has_positions:
+            return False
+        if self.has_positions:
+            if not (self.positions == other.positions).all():
+                return False
+
+        if self.dimensions is None:
+            if other.dimensions is not None:
+                return False
+        else:
+            if other.dimensions is None:
+                return False
+            if not (self.dimensions == other.dimensions).all():
+                return False
+
+        if not self.has_velocities == other.has_velocities:
+            return False
+        if self.has_velocities:
+            if not (self.velocities == other.velocities).all():
+                return False
+
+        if not self.has_forces == other.has_forces:
+            return False
+        if self.has_forces:
+            if not (self.forces == other.forces).all():
+                return False
+
+        return True
+
+    def __ne__(self, other):
+        return not self == other
```

Comment:
I can't remember if this is automatically implemented in py3?

Reply:
Defers to `__eq__` and inverts so I have removed.